frontend/src/dataloaders/loader_helpers/SharedFunctions.ts

```diff
@@ -74,12 +74,26 @@ export interface projectsAndSubjects {
 }
 
 export async function getAllProjectsAndSubjects(role: teacherStudentRole, filter_on_current: boolean = false): Promise<projectsAndSubjects> {
-    const apiSubjects = (await apiFetch(`/${role}/subjects`)) as Backend_Subject[];
-    const apiProjects = (await apiFetch(`/${role}/projects`)) as Backend_Project[];
-    let projects: Project[] = mapProjectList(apiProjects);
-    if (filter_on_current) {
-        projects = projects.filter(project => project.project_visible && !project.project_archived)
+    const apiProjects = (await apiFetch<Backend_Project[]>(`/${role}/projects`));
+    let projects: Project[] = []
+
+    if (apiProjects.ok) {
+        projects = mapProjectList(apiProjects.content);
+        if (filter_on_current) {
+            projects = projects.filter(project => project.project_visible && !project.project_archived)
+        }
+    }else{
+        // TODO error handling
+    }
```

Comment:
Als je de arg van je if zou inverten dan zou je dit wat mooier kunnen schrijven doordat je projects dan een const kunt maken en direct kunt = ... doen.
bvb
```ts
if (!apiProjects.ok) {
    // TODO error handling
} 
// vanaf hier weet je dat het toch werkt.
const projects = mapProjectList(...)
...
```
je zou zelf mss de ok's samen kunnen checken (je hebt dan ook 2 promises naast elkaar lopen wat mss iets sneller is.) dit zou voor 1 plaats zorgen waar je errors kan aanpakken.

Reply:
Dat is inderdaad mooier, ik heb het aangepast met de error handling in een TODO zodat dat kan worden bekeken als de foutmeldingen geïmplementeerd worden.